### PR TITLE
[HomeArtworkRail] Rail layout for initial state

### PR DESCRIPTION
**Pull request by sarahscott**
This is a preliminary go at a nice way to do the initial state. I will follow up soon with a final state and my first go at animating between the two.

expansionanimation

### CODE REVIEW

File: lib/components/home/artwork_rail.js [hunk 1]
```diff
@@ -3,22 +3,72 @@
...
+  }
+
   renderModuleResults() {
-    if (this.props.rail.results) {
+    if (this.props.rail.results && this.props.rail.results.length) {
       return (
-        <Grid artworks={this.props.rail.results}/>
+        <View style={[styles.container, { height: this.state.expanded ? 2350 : 500 }]}>
+          <View style={[styles.gridContainer, { height: this.state.expanded ? 2300 : 420 }]}>
+            <Grid artworks={this.props.rail.results}/>
```
Comment (sarahscott) [inline lib/components/home/artwork_rail.js]: If I add an `onLayout` prop here and bind it to some function in this rail, it is never called. If I add `onLayout` to any other view, it behaves as expected and is called upon layout. I'm not sure how to get the height of the grid when it's finished laying itself out, and that value would be useful in the above lines for setting the expanded height of the `container` and `gridContainer`.

**Comment by orta** [comment]
dependency-ci  is down, we're seeing it on eigen too

File: lib/components/home/artwork_rail.js (hunk 1 above)
Comment (alloy) [inline lib/components/home/artwork_rail.js]: What happens when you wrap the Grid in another wrapper and set the `onLayout` prop there?

File: lib/components/home/artwork_rail.js [hunk 2]
```diff
@@ -3,22 +3,72 @@
...
+    return null;
+  }
+
+  onViewAllPress() {
+    // pop new view controller
+  }
+
   renderModuleResults() {
-    if (this.props.rail.results) {
+    if (this.props.rail.results && this.props.rail.results.length) {
```
Comment (alloy) [inline lib/components/home/artwork_rail.js]: I think we already pull lodash in, so we could start using [isEmpty](https://lodash.com/docs#isEmpty) for these things.

File: lib/components/home/artwork_rail.js [hunk 3]
```diff
@@ -3,22 +3,72 @@
...
+            <Grid artworks={this.props.rail.results}/>
+            { this.renderViewAllButton() }
+          </View>
+          <Separator/>
+          { this.renderExpansionButton() }
+        </View>
       );
+    } else if (this.props.rail.results) {
+      // temporary; for those pesky empty rails on staging
+      return <View/>;
```
Comment (alloy) [inline lib/components/home/artwork_rail.js]: Which rails are those?

**Comment by alloy** [comment]
w00t, looks amaze :fire:

**Pull request merged**